packages/storage/src/database/global/proof.ts

```diff
@@ -41,11 +42,43 @@ export class ModelProof extends ModelGeneral<WithoutId<TProofRecord>> {
   public async getProof(
     database: string,
     options?: FindOptions
-  ): Promise<WithId<TProofRecord> | null> {
-    const proof = await this.collection.findOne(
+  ): Promise<TProofRecord | null> {
+    return this.collection.findOne(
       { database },
       { ...options, sort: { createdAt: -1 } }
     );
-    return proof as WithId<TProofRecord> | null;
+  }
+
+  public static async init(session?: ClientSession) {
+    const collection = ModelCollection.getInstance<TProofRecord>(
+      zkDatabaseConstant.globalProofDatabase,
+      DB.proof,
+      zkDatabaseConstant.globalCollection.proof
+    );
+    /*
+      publicInput: string[];
+      publicOutput: string[];
+      maxProofsVerified: 0 | 1 | 2;
+      proof: string;
+      createdAt?: Date;
+      database: string;
+      collection: string;
+      merkleRoot: string;
+      prevMerkleRoot: string;
+    */
+    if (!(await collection.isExist())) {
+      await collection.index({ proof: 1 }, { unique: true, session });
+      await collection.index({ database: 1 }, { session });
```

Comment:
shouldn't this be databaseName?
https://github.com/orochi-network/zkDatabase/blob/463aa241de15594dabbdfb0695bc8d7bb75f78e8/packages/common/src/types/proof.ts#L73

Reply:
resolved. it databaseName and collectionName